ext/libxml/ruby_xml_node.c

```diff
@@ -45,13 +45,18 @@ static void rxml_node_deregisterNode(xmlNodePtr xnode)
   /* Has the node been wrapped and exposed to Ruby? */
   if (xnode->_private)
   {
+    /* Does it belong to another libxml user? */
+    if (!OWNED(xnode))
```

Comment:
If we go down this route, why set _private at all?  Just save the mapping in the hash table between the libxml pointer and the created ruby object and never touch _private.  libxml-ruby used _private because it was convenient, and the libxml documents encourage it:

"Most nodes also carry an _private pointer which can be used by the application to hold specific data on this node."

And of course Nokogiri didn't exist back then either, so it was a non-issue.  If we are going to change this, let's totally move away from _private if possible.